hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/VirtualIdEmbeddable.java

```diff
@@ -247,6 +253,61 @@ private boolean finishInitialization(
 		);
 	}
 
+	@Override
+	protected boolean initColumnMappings() {
+		final List<SelectableMapping> selectableMappings = CollectionHelper.arrayList( attributeMappings.size() );
+		final SelectableConsumer selectableConsumer = (columnIndex, selection) -> selectableMappings.add( selection );
+		attributeMappings.forEach( attributeMapping -> {
+			if ( attributeMapping instanceof ToOneAttributeMapping
+					&& ( (ToOneAttributeMapping) attributeMapping ).getSideNature() != ForeignKeyDescriptor.Nature.KEY ) {
+				/*
+				    We need to consider the associated type's identifier selectable mappings even on the TARGET side
+				    as they will still be part of the composite primary key for this virtual embeddable.
+					See the following mapping as an example:
+
+					@Entity
+					@IdClass( ProductPK.class )
+					class Product {
+						@Id
+						private String productId;
+						@OneToOne( mappedBy = "product" )
+						private Price wholesalePrice;
+					}
+
+					class ProductPK {
+						private String productId;
+						@Embedded
+						private PricePK wholesalePrice;
+					}
+
+					@Entity
+					@IdClass( PricePK.class )
+					class Price {
+						@Id
+						private Integer operatorId;
+						@Id
+						private String wholesalePrice;
+						@OneToOne
+						private Product product;
+					}
+
+					class PricePK {
+						private Integer operatorId;
+						private String wholesalePrice;
+					}
+				 */
+				( (ToOneAttributeMapping) attributeMapping ).getAssociatedEntityMappingType()
+						.getIdentifierMapping()
+						.forEachSelectable( selectableConsumer );
```

Comment:
Well, not really, because the target side is the same as the `VirtualIdEmbeddable`'s container, so it wouldn't be correct: the selectables in that case would be the composite ID of the current entity, while we want to create a primary key with the properties of the associated type's composite identifier. That is the behavior we observed in Hibernate 5.